### PR TITLE
ADD pause subscriptions using the NGSIv2 API and a couple of csubs cache fixes

### PR DESCRIPTION
**Pull request by fgalan**
Implements #1328 and a couple of fixed in csubs cache that I have need to introduce on the way.

@kzangeli 
@crbrox

### CODE REVIEW

File: src/lib/mongoBackend/mongoSubCache.cpp [hunk 1]
```diff
@@ -466,8 +468,11 @@ void mongoSubCacheUpdate(const std::string& tenant, const std::string& subId, lo
 
   if (lastNotificationTime != 0)
   {
-    // Update lastNotificationTime
-    condition = BSON("_id" << OID(subId) << CSUB_LASTNOTIFICATION << BSON("$lt" << lastNotificationTime));
+    // Update lastNotificationTime    
+    condition = BSON("_id" << OID(subId) << "$or" << BSON_ARRAY(
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoSubCache.cpp]: This is the fix for:

"Fix: csub cache sync logic not persisting lastNotification when the field doesn't initially exist in DB"

File: src/lib/mongoBackend/mongoUpdateContextSubscription.cpp [hunk 2]
```diff
@@ -174,32 +174,26 @@ HttpStatusCode mongoUpdateContextSubscription
...
 
-  /* Duration update */
-  long long expiration = -1;
-
-  if (requestP->expires != -1) //v2
-  {
-    expiration = requestP->expires;
-  }
-  else
+  /* Expiration */
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoUpdateContextSubscription.cpp]: This is the fix for:

"Fix: csub cache at mongoBackend update context subscription logic wrongly setting expiration with current time, ignoring actual expires/duration in the request"

File: test/functionalTest/cases/1048_subscription_cache/csub_update_with_notif.test [hunk 3]
```diff
@@ -0,0 +1,447 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (fgalan) [inline test/functionalTest/cases/1048_subscription_cache/csub_update_with_notif.test]: Note this .test is in the case/1048_XXXX (not in cases/1328_XXX), due to is a test to check the ""Fix: csub cache sync logic not persisting lastNotification when the field doesn't initially exist in DB".

File: src/lib/jsonParseV2/parseSubscription.cpp [hunk 4]
```diff
@@ -171,6 +171,33 @@ std::string parseSubscription(ConnectionInfo* ciP, ParseData* parseDataP, JsonDe
...
+
+      alarmMgr.badInput(clientIp, "status is not a string");
+      return oe.render(ciP, "");
+    }
+
+    std::string statusString = status.GetString();
+
+    if ((statusString != "active") && (statusString != "inactive"))
+    {
+      OrionError oe(SccBadRequest, "status is not valid (it has to bee either active or inactive)");
```
Comment (kzangeli) [inline src/lib/jsonParseV2/parseSubscription.cpp]: typo: bee -> be (in the alarm as well)

File: test/functionalTest/cases/1048_subscription_cache/csub_update_with_notif.test [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+#
+# Note we use sleep in some point to let time the DB to sync with regards to
+# cache timeSent value (WAIT_TIME = 1.5 * subCacheIval)
+#
+WAIT_TIME=1.5s
+#
+# 01. Create subscription
+# 02. Create entity (triggers notification)
+# 03. Check subscription timesSent value (1)
+# 04. Subcription update not involing actual update
```
Comment (kzangeli) [inline test/functionalTest/cases/1048_subscription_cache/csub_update_with_notif.test]: typo: involing -> involving (in 6 places)

Comment (fgalan) [inline test/functionalTest/cases/1048_subscription_cache/csub_update_with_notif.test]: Fixed in f767db6090bf924f628be0b2a415477a77f1cbfa

File: src/lib/jsonParseV2/parseSubscription.cpp (hunk 4 above)
Comment (fgalan) [inline src/lib/jsonParseV2/parseSubscription.cpp]: Fixed in f767db6090bf924f628be0b2a415477a77f1cbfa

**Comment by kzangeli** [comment]
LGTM

**Comment by crbrox** [comment]
LGTM

**Pull request merged**